cmd/gaia/app/genesis.go

```diff
@@ -155,20 +155,42 @@ func NewDefaultGenesisState() GenesisState {
 // TODO: No validators are both bonded and jailed (#2088)
 // TODO: Error if there is a duplicate validator (#1708)
 // TODO: Ensure all state machine parameters are in genesis (#1704)
-func GaiaValidateGenesisState(genesisState GenesisState) (err error) {
-	err = validateGenesisStateAccounts(genesisState.Accounts)
+func GaiaValidateGenesisState(genesisState GenesisState) error {
+	err := validateGenesisStateAccounts(genesisState.Accounts)
 	if err != nil {
-		return
+		return err
 	}
 	// skip stakeData validation as genesis is created from txs
 	if len(genesisState.GenTxs) > 0 {
 		return nil
 	}
-	return stake.ValidateGenesis(genesisState.StakeData)
+
+	stake.ValidateGenesis(genesisState.StakeData)
+	if err != nil {
+		return err
+	}
+	mint.ValidateGenesis(genesisState.MintData)
+	if err != nil {
+		return err
+	}
+	distr.ValidateGenesis(genesisState.DistrData)
+	if err != nil {
+		return err
+	}
+	gov.ValidateGenesis(genesisState.GovData)
+	if err != nil {
+		return err
+	}
+	slashing.ValidateGenesis(genesisState.SlashingData)
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
 // Ensures that there are no duplicate accounts in the genesis state,
-func validateGenesisStateAccounts(accs []GenesisAccount) (err error) {
```

Comment:
Are we looking to remove this style of returns?

Reply:
The general consensus from conversations is to always include a return argument _at_ each return statement, and, whenever necessary _also_ include an output field. Examples where this might be necessary would be: a string output where the string actually represents something more detailed such as a key; or some raw.json which represents a specific object; etc.  Here we are not gaining any additional valuable information by simply labelling the `error` as `err` it's already obvious what `error` is.

Reply:
++